pysatCDAAC/instruments/cosmic2_ivm.py

```diff
@@ -0,0 +1,160 @@
+
+_test_dates = {inst_id: {'': dt.datetime(2021, 1, 1)}
+               for inst_id in inst_ids.keys()}
+
+# ----------------------------------------------------------------------------
+# Instrument methods
+
+
+def init(self):
+    """Initialize the Instrument object with instrument specific values.
+
+    Note
+    ----
+    Runs once upon instantiation.
+
+    """
+    ack = ' '.join(('FORMOSAT-7/COSMIC-2 is a partnership between the National',
+                    'Space Organization in Taiwan and NOAA, the U.S. Air',
+                    'Force, and the University Corporation for Atmospheric',
+                    'Research (UCAR) in the United States.'))
+    refs = ' '.join(('UCAR COSMIC Program, 2019: COSMIC-2 Data Products [Data',
+                     'set]. UCAR/NCAR - COSMIC, Access date [insert date],'
+                     'https://doi.org/10.5065/T353-C093'))
+    self.acknowledgements = ack
+    self.references = refs
+    pysat.logger.info(ack)
+    pysat.logger.info(' '.join(('All spacecraft data stored in single tar',
+                                'files on the CDAAC website. Pysat will',
+                                'download all and update accordingly.')))
+
+    return
+
+
+def clean(self):
+    """Clean COSMIC2 IVM data to the specified level.
+
+    Note
+    ----
+    'clean' - Not specified
+    'dusty' - Not specified
+    'dirty' - Not specified
+    'none'  No cleaning applied, routine not called in this case.
+
+    """
+    warnings.warn('Cleaning not yet implemented')
+
+    return
+
+
+def load(fnames, tag='', inst_id=''):
+    """Load COSMIC2 IVM data into `pandas.DataFrame` and `pysat.Meta` objects.
+
+    This routine is called as needed by pysat. It is not intended
+    for direct user interaction.
+
+    Parameters
+    ----------
+    fnames : array-like
+        iterable of filename strings, full path, to data files to be loaded.
+        This input is nominally provided by pysat itself.
+    tag : str
+        tag name used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself. (default='')
+    inst_id : str
+        Satellite ID used to identify particular data set to be loaded.
+        This input is nominally provided by pysat itself. (default='')
+
+    Returns
+    -------
+    data : pds.DataFrame
+        A pandas DataFrame with data prepared for the pysat.Instrument
+    meta : pysat.Meta
+        Metadata formatted for a pysat.Instrument object.
+
+    Note
+    ----
+    Any additional keyword arguments passed to pysat.Instrument
+    upon instantiation are passed along to this routine.
+
+    Examples
+    --------
+    ::
+
+        inst = pysat.Instrument('cosmic2', 'ivm', inst_id='e1')
+        inst.load(2020, 1)
+
+    """
+
+    data, meta = pysat.utils.io.load_netcdf(
+        fnames, epoch_name='time', epoch_unit='s',
+        epoch_origin=dt.datetime(1980, 1, 6), 
+        meta_translation={'_FillValue': meta.labels.fill_val})
```

Comment:
@rstoneback, `meta` isn't defined yet.  Not sure what should go here instead.

Reply:
Should I invoke `meta = pysat.Meta()` to grab default values?

Reply:
oh, for this case `meta.labels.fill_val` should be what is actually in the file.

Reply:
I mean, hardcode the value that is in the file.

Reply:
There is no fill value in the file.  I could just use 'fill_val'.  I'm not getting any errors when I run this without the meta_translation.  Could something have changed in the core pysat?

Reply:
That feedback was pushed to debug.
```
In [13]: ivm.load(2020, 1)
pysat DEBUG: Translation label "_FillValue" not found for variable "frac_dens_h".
pysat DEBUG: Translation label "_FillValue" not found for variable "frac_dens_o".
pysat DEBUG: Translation label "_FillValue" not found for variable "frac_dens_ion4".
pysat DEBUG: Translation label "_FillValue" not found for variable "frac_dens_ion5".
pysat DEBUG: Translation label "_FillValue" not found for variable "ion_dens".
pysat DEBUG: Translation label "_FillValue" not found for variable "orig_ion_density".
pysat DEBUG: Translation label "_FillValue" not found for variable "ion_temp".
pysat DEBUG: Translation label "_FillValue" not found for variable "ap_pot".
pysat DEBUG: Translation label "_FillValue" not found for variable "uts".
pysat DEBUG: Translation label "_FillValue" not found for variable "sc_flag".
pysat DEBUG: Translation label "_FillValue" not found for variable "rpa_flag".
pysat DEBUG: Translation label "_FillValue" not found for variable "dm_flag".
pysat DEBUG: Translation label "_FillValue" not found for variable "ion_temp_var".
```

Reply:
Inspecting the files with xarray, I don't see fill values specified.  Not sure if we should put in a fake value to satisfy the debug statement.

Reply:
I don't see fill values when loading direct from netCDF4 either. CDAAC folks specified what they wanted in the file consistent with their netCDF3 standard. If you update the meta_translation value in the load statement, like meta_translation={}, then it should be ok. Looks ok on my system.